src/BuildScriptGenerator/DotNetCore/DefaultAspNetCoreWebAppProjectFileProvider.cs

```diff
@@ -171,6 +169,25 @@ internal static bool IsAspNetCoreWebApplicationProject(XDocument projectFileDoc)
             return string.Equals(sdkName, expectedWebSdkName, StringComparison.OrdinalIgnoreCase);
         }
 
+        // To enable unit testing
+        internal static string GetRelativePathToRoot(string projectFilePath, string repoRoot)
+        {
+            var repoRootDir = new DirectoryInfo(repoRoot);
+            var projectFileInfo = new FileInfo(projectFilePath);
+            var currDir = projectFileInfo.Directory;
+            var parts = new List<string>();
+            parts.Add(projectFileInfo.Name);
+
+            // Since directory names are case sensitive on non-Windows OSes, try not to use ignore case
+            while (!string.Equals(currDir.FullName, repoRootDir.FullName, StringComparison.Ordinal))
```

Comment:
I'm concerned that this might throw an exception if those will never be equal, e.g. when `projectFilePath` is not in a subdir of `repoRoot`... I think we should handle this error case here just to make sure it fails gracefully if that ever happens